corehq/apps/sms/views.py

```diff
@@ -665,36 +665,110 @@ def global_backend_map(request):
     }
     return render(request, "sms/backend_map.html", context)
 
+
 @require_permission(Permissions.edit_data)
 @requires_privilege_with_fallback(privileges.OUTBOUND_SMS)
 def chat_contacts(request, domain):
-    domain_obj = Domain.get_by_name(domain, strict=True)
-    verified_numbers = VerifiedNumber.by_domain(domain)
-    contacts = []
-    for vn in verified_numbers:
-        owner = vn.owner
-        if owner is not None and owner.doc_type in ('CommCareCase','CommCareUser'):
-            if owner.doc_type == "CommCareUser":
-                url = reverse(EditCommCareUserView.urlname, args=[domain, owner._id])
-                name = owner.raw_username
-            else:
-                url = reverse("case_details", args=[domain, owner._id])
-                if domain_obj.custom_case_username:
-                    name = owner.get_case_property(domain_obj.custom_case_username) or _("(unknown)")
-                else:
-                    name = owner.name
-            contacts.append({
-                "id" : owner._id,
-                "doc_type" : owner.doc_type,
-                "url" : url,
-                "name" : name,
-            })
     context = {
         "domain" : domain,
-        "contacts" : contacts,
     }
     return render(request, "sms/chat_contacts.html", context)
 
+
+def get_case_contact_info(domain_obj, case_ids):
+    data = {}
+    for doc in iter_docs(CommCareCase.get_db(), case_ids):
+        if domain_obj.custom_case_username:
+            name = doc.get(domain_obj.custom_case_username, _('unknown'))
+        else:
+            name = doc.get('name', _('unknown'))
+        data[doc['_id']] = [name]
+    return data
+
+
+def get_mobile_worker_contact_info(domain_obj, user_ids):
+    data = {}
+    for doc in iter_docs(CommCareUser.get_db(), user_ids):
+        user = CommCareUser.wrap(doc)
+        data[user.get_id] = user.raw_username
```

Comment:
This value is set to a string, but for cases the value is set to a list. When you pull out contact_info[0] in line 725, I think you will only get the first character of raw_username. 

Should this be `data[user.get_id] = [user.raw_username]`?